public/js/clients/firefox.js

```diff
@@ -146,6 +146,45 @@ function createFrame(frame) {
 }
 
 const CALL_STACK_PAGE_SIZE = 25;
+const NEW_SOURCE_IGNORED_URLS = ["debugger eval code", "XStringBundle"];
+
+function makeDispatcher(client, actions) {
+  return {
+    paused(_, packet) {
+      // If paused by an explicit interrupt, which are generated by the
+      // slow script dialog and internal events such as setting
+      // breakpoints, ignore the event.
+      if (packet.why.type === "interrupted" && !packet.why.onNext) {
+        return;
+      }
+
+      // Eagerly fetch the frames
+      client.getFrames(0, CALL_STACK_PAGE_SIZE, res => {
+        actions.loadedFrames(res.frames.map(createFrame));
+      });
```

Comment:
Not sure what you mean; loading the frames is different than the sources. The main difference in here is at the bottom of `initPage` we call `client.getSources` instead of `actions.loadSources`. We were doing this before too; I just lumped in this event dispatcher stuff in here as well, but it just moved the previous event handlers up into here.